core/modules/user/user.module

```diff
@@ -1772,13 +1791,49 @@ function user_page_title($account) {
   return is_object($account) ? user_format_name($account) : '';
 }
 
+/**
+ * Implements hook_init().
+ */
+function user_init() {
+  $current_path = current_path();
+  $user_login_page_settings = user_login_page_settings();
+  if (!user_is_logged_in() && ($current_path == 'user/login' || $current_path == 'user/password' || $current_path == 'user/register')) {
+    if ($user_login_page_settings == 'simplified') {
+      layout_suppress(TRUE);
+    }
+  }
+}
+
+/**
+ * Implements hook_custom_theme().
+ */
+function user_custom_theme() {
+  $current_path = current_path();
+  $user_login_page_settings = user_login_page_settings();
+  if ($user_login_page_settings == 'simplified' && !user_is_logged_in() && ($current_path == 'user/login' || $current_path == 'user/password' || $current_path == 'user/register')) {
+    return config_get('system.core', 'admin_theme');
+  }
+}
+
+/**
+ * Implements hook_preprocess_page().
+ */
+function user_preprocess_page(&$variables) {
+  // Add a class that tells us whether we're on a login page or not.
+  $current_path = current_path();
+  if ($current_path == 'user/login' || $current_path == 'user/password' || $current_path == 'user/register') {
+    $variables['classes'][] = 'user-login-page';
```

Comment:
If we do the other renaming, this could change to:
```suggestion
    $variables['classes'][] = 'user-simplified-page';
```

Because I think that's the only situation where this is needed. The class shouldn't be added on the tabs or link versions (which don't need it).

Reply:
Confusion resolved, hopefully, by renaming things a bit.